modules/apps/headless/headless-discovery/headless-discovery-impl/src/main/java/com/liferay/headless/discovery/internal/jaxrs/application/HeadlessDiscoveryOpenAPIApplication.java

```diff
@@ -70,7 +70,8 @@ public Map<String, List<String>> openAPI(
 		Map<String, List<String>> pathsMap = new TreeMap<>();
 
 		String serverURL =
-			_portal.getPortalURL(_httpServletRequest) + Portal.PATH_MODULE;
+			_portal.getPortalURL(_httpServletRequest) + _portal.getPathProxy() +
+				Portal.PATH_MODULE;
```

Comment:
I think we modified something like that already in the _UriInfoUtil_ class. In order to avoid duplicating way of creating URLs, could we do it like this?:

`StringUtil.removeLast(UriInfoUtil.getBasePath(_uriInfo), "/openapi/")`

Or it fails?

Reply:
After testing, can confirm that only way to get schema and host information of a reverse-proxy server is through HTTPServletRequest and its headers X-Forwarded-Proto and X-Forwarded-Host. 

UriInfo will contain only information about request that was created from reverse-proxy to LR instance. For example in the following case where HTTPS communication is terminated at reverse-proxy

 **client** ----https://liferay.com---- **reverse-proxy** ----http://clientDoesn'tSeeThis.com---- **LR instance** 

UriInfo will have HTTP protocol and clientDoesn'tSeeThis host defined, which will generate wrong URLs for openAPI explorer.

That being said, I agree we should prettify the way we're generating this URLs and not have that logic inlined inside the method, but really nothing comes to mind at the moment!